bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/type/HomeConnectDynamicStateDescriptionProvider.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.homeconnect.internal.type;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.type.DynamicStateDescriptionProvider;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.openhab.binding.homeconnect.internal.logger.EmbeddedLoggingService;
+import org.openhab.binding.homeconnect.internal.logger.LogWriter;
+import org.openhab.binding.homeconnect.internal.logger.Type;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.event.Level;
+
+/**
+ * The {@link HomeConnectDynamicStateDescriptionProvider} is responsible for handling dynamic thing values.
+ *
+ * @author Jonas Brüstel - Initial contribution
+ */
+@Component(service = { DynamicStateDescriptionProvider.class, HomeConnectDynamicStateDescriptionProvider.class })
+@NonNullByDefault
+public class HomeConnectDynamicStateDescriptionProvider implements DynamicStateDescriptionProvider {
+
+    private final ConcurrentHashMap<String, StateDescription> stateDescriptions = new ConcurrentHashMap<>();
+    private final LogWriter logger;
+
+    @Activate
+    public HomeConnectDynamicStateDescriptionProvider(@Reference EmbeddedLoggingService loggingService) {
+        logger = loggingService.getLogger(HomeConnectDynamicStateDescriptionProvider.class);
+    }
+
+    @Override
+    public @Nullable StateDescription getStateDescription(Channel channel,
+            @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
+        if (stateDescriptions.containsKey(channel.getUID().getAsString())) {
+            logger.log(Type.DEFAULT, Level.DEBUG, null, null,
+                    Arrays.asList(stateDescriptions.get(channel.getUID().getAsString()).toString()), null, null,
+                    "Return dynamic state description for channel-uid {}.", channel.getUID().getAsString());
+            return stateDescriptions.get(channel.getUID().getAsString());
+        }
+
+        return originalStateDescription;
```

Comment:
You should not return the originalStateDescription here. It will break other bindings that have dynamic state descriptions.

See [here](https://community.openhab.org/t/dynamic-state-options-not-showing-up/95625) and [here](https://community.openhab.org/t/squeezebox-favorites/39355/69?u=mhilbush) for more information.

Return null instead.

Reply:
Thanks for the information. I will change it.

Reply:
And another one bites the dust: https://community.openhab.org/t/heos-denon-support/3449/322?u=martinvw